bin/runtime/src/lib.rs

```diff
@@ -750,6 +751,84 @@ impl pallet_identity::Config for Runtime {
     type RegistrarOrigin = EnsureRoot<AccountId>;
     type WeightInfo = pallet_identity::weights::SubstrateWeight<Self>;
 }
+parameter_types! {
+    // Key size = 32, value size = 8
+    pub const ProxyDepositBase: Balance = 40 * LEGACY_DEPOSIT_PER_BYTE;
+    // One storage item (32) plus `ProxyType` (1) encode len.
+    pub const ProxyDepositFactor: Balance = 33 * LEGACY_DEPOSIT_PER_BYTE;
+    // Key size = 32, value size 8
+    pub const AnnouncementDepositBase: Balance =  40 * LEGACY_DEPOSIT_PER_BYTE;
+    // AccountId, Hash and BlockNumber sum up to 68
+    pub const AnnouncementDepositFactor: Balance =  68 * LEGACY_DEPOSIT_PER_BYTE;
+}
+#[derive(
+    Copy,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Encode,
+    Decode,
+    RuntimeDebug,
+    MaxEncodedLen,
+    scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+    Any = 0,
+    NonTransfer = 1,
+    Staking = 2,
+}
+impl Default for ProxyType {
+    fn default() -> Self {
+        Self::Any
+    }
+}
+impl InstanceFilter<RuntimeCall> for ProxyType {
+    fn filter(&self, c: &RuntimeCall) -> bool {
+        match self {
+            ProxyType::Any => true,
+            ProxyType::NonTransfer => !matches!(
+                c,
+                RuntimeCall::Balances(..)
+                    | RuntimeCall::Vesting(pallet_vesting::Call::vested_transfer { .. })
+            ),
```

Comment:
This is quite dangerous. If we ever add a pallet that can do any type of transfers we must remember to update this... Even now, I think this is incorrect. Because you can make transfers using `Contract` calls. 

We need a plan on how to make this safe.

Reply:
In general any proxy that works by blacklisting, instead of whitelisting calls is dangerous.

Reply:
There is also an option to not add the non-trasnfer proxy

Reply:
I think it's best to define it via whitelisting...

Reply:
And no "capture-all" statements - if we add a new variant we will get a compilation error, rather than a runtime surprise.

Reply:
What about we do same as per 
https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/lib.rs#L960-L998 ?

Reply:
Sorry, this is link to the archived repo, but we can check how polkadot-sdk repo does it

Reply:
Ok, non-transfer changed to 'whitelisted' variant